mmh3.go

```diff
@@ -199,160 +186,190 @@ func Hash128x64(key []byte) []byte {
 	return Hash128(key).Bytes()
 }
 
-// WriteHash128x64 is a
+// WriteHash128x64 creates a hash of key and writes it to ret
 func WriteHash128x64(key, ret []byte) {
 	Hash128(key).Write(ret)
 }
 
 type HashWriter128 struct {
-	buf     [16]byte
-	h1, h2  uint64
-	index   int
-	written int64
+	h1         uint64
+	h2         uint64
+	tail       [h128BlockSize]byte
+	tailLength int
+	length     int
 }
 
 func (hw *HashWriter128) Reset() {
-	hw.buf = [16]byte{}
```

Comment:
does `hw.tail` need to be reset here?

Reply:
No.  The `tailLength` value controls how many bytes of the tail that we look at and we reset that to zero.  The next call to `AddBytes` will update the tail and length if needed.